llm/llama.py

```diff
@@ -0,0 +1,76 @@
+import requests
+import json
+
+class Settings:
+    llama_host = "http://localhost:8080"
+    llama_prompt = ("You are an intelligent and polite medical assistant "
+                    "who provides detailed and "
+                    "helpful answers to user's medical questions, "
+                    "including accurate references where applicable.")
+
+settings = Settings()
+
+class LlamaCppClient:
+    def __init__(self, base_url=settings.llama_host):
+        self.base_url = base_url
+        self.llama_prompt = settings.llama_prompt
+        self.conversation_history = []
+
+    def chat(self, message, params=None):
+        if params is None:
+            params = {
+                "n_predict": 400,
+                "temperature": 0.7,
```

Comment:
From where did you get this value? Is this not too high? I believe a temperature closer to 1 will produce more hallucinations

Reply:
Default value is 0.8: https://github.com/ggerganov/llama.cpp/blob/master/examples/server/README.md, we could reduce this value in the generation tests